packages/website/components/MetamaskNetworkButton/ConnectToMetamask.tsx

```diff
@@ -0,0 +1,87 @@
+type Props = {
+  buttonText: string;
+};
+
+async function ConnectToMetamask(props: Props) {
+  if (!(window as any).ethereum) {
+    alert("Wallet not detected! Install Metamask then try again.");
+    return;
+  }
+  if(props.buttonText == "Connect to Sepolia"){
+    ConnectToSepolia();
+  }
+  if(props.buttonText == "Connect to Taiko"){
+    ConnectToTaiko();
+  }
+}
+
+async function ConnectToSepolia() {
+  if (!(window as any).ethereum) {
+    alert("Metamask not detected! Install Metamask then try again.")
+    return;
+  }
+  if ((window as any).ethereum.networkVersion == "11155111") {
+    alert("You are already connected to Sepolia (chainId 11155111).", )
+    return;
+  }
+  try{
+    await (window as any).ethereum.request({
+      method: "wallet_switchEthereumChain",
+      params: [{
+         chainId: "0xaa36a7"
```

Comment:
curious will this work if they have not enabled the test networks in metamask?

Reply:
You switch networks instead of adding them if they already exist in Metamask by default for security purposes:
https://ethereum.stackexchange.com/questions/127499/metamask-not-switching-chain-to-rinkeby-testnet

Reply:
This also works in Metamask when testnets like Sepolia are hidden based on a test I did now.